### ISSUE TITLE
[Windows] Failed to launch window example on AMD GPU

### ISSUE THREAD

**Issue by kchibisov** [rust-windowing/glutin#1219]
Windows Version: `Windows 10 (64 bit)`

Radeon Software Version: `19.5.2`

Windows is running natively.
GPU: `Rx 570`

`cargo run --example window` fails with
```
$ RUST_BACKTRACE=1 cargo run --example window                                                                                                 Finished dev [unoptimized + debuginfo] target(s) in 0.07s
     Running `target\debug\examples\window.exe`
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: NoAvailablePixelFormat', src\libcore\result.rs:999:5
stack backtrace:
   0: backtrace::backtrace::trace_unsynchronized
             at C:\Users\VssAdministrator\.cargo\registry\src\github.com-1ecc6299db9ec823\backtrace-0.3.29\src\backtrace\mod.rs:66
   1: std::sys_common::backtrace::_print
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\sys_common\backtrace.rs:47
   2: std::sys_common::backtrace::print
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\sys_common\backtrace.rs:36
   3: std::panicking::default_hook::{{closure}}
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panicking.rs:200
   4: std::panicking::default_hook
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panicking.rs:214
   5: std::panicking::rust_panic_with_hook
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panicking.rs:477
   6: std::panicking::continue_panic_fmt
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panicking.rs:384
   7: std::panicking::rust_begin_panic
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panicking.rs:311
   8: core::panicking::panic_fmt
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libcore\panicking.rs:85
   9: core::result::unwrap_failed<glutin::CreationError>
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\src\libcore\macros.rs:18
  10: core::result::Result<glutin::windowed::ContextWrapper<glutin::context::NotCurrent, winit::window::Window>, glutin::CreationError>::unwrap<glutin::windowed::ContextWrapper<glutin::context::NotCurrent, winit::window::Window>,glutin::CreationError>
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\src\libcore\result.rs:800
  11: window::main
             at .\glutin_examples\examples\window.rs:13
  12: std::rt::lang_start::{{closure}}<()>
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\src\libstd\rt.rs:64
  13: std::rt::lang_start_internal::{{closure}}
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\rt.rs:49
  14: std::panicking::try::do_call<closure,i32>
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panicking.rs:296
  15: panic_unwind::__rust_maybe_catch_panic
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libpanic_unwind\lib.rs:82
  16: std::panicking::try
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panicking.rs:275
  17: std::panic::catch_unwind
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\panic.rs:394
  18: std::rt::lang_start_internal
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\/src\libstd\rt.rs:48
  19: std::rt::lang_start<()>
             at /rustc/eae3437dfe991621e8afdc82734f4a172d7ddf9b\src\libstd\rt.rs:64
  20: main
  21: invoke_main
             at d:\agent\_work\3\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:78
  22: __scrt_common_main_seh
             at d:\agent\_work\3\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288
  23: BaseThreadInitThunk
  24: RtlUserThreadStart
error: process didn't exit successfully: `target\debug\examples\window.exe` (exit code: 101)
```

This issue doesn't exist on glutin `v0.21.0`, however `alacritty` on evlp2 starts kind of normally (it spawns one tmp window for whatever reason, but then everything is fine).

**Comment by ZeGentzy** [comment]
Can you follow the instructions provided here: https://github.com/rust-windowing/glutin/issues/952#issuecomment-467228004

**Comment by kchibisov** [comment]
Let me know, if you need more information or a different one.

```
Renderer: Radeon RX 570 Series
Vendor: ATI Technologies Inc.
Memory: 8192 MB
Version: 4.6.13558 Compatibility Profile Context 26.20.11015.5009
Shading language version: 4.60


Max texture size: 16384 x 16384
Max texture coordinates: 0
Max vertex texture image units: 32
Max texture image units: 32
Max geometry texture units: 32
Max anisotropic filtering value: 16
Max number of light sources: 8
Max viewport size: 16384 x 16384
Max uniform vertex components: 16384
Max uniform fragment components: 16384
Max geometry uniform components: 16384
Max varying vectors: 128
Max samples: 8
Max draw buffers: 8


GL Extensions: 323
 

GL_ARB_tessellation_shader
GL_ARB_parallel_shader_compile
GL_ARB_half_float_pixel
GL_EXT_compiled_vertex_array
GL_AMD_shader_stencil_value_export
GL_WIN_swap_hint
GL_EXT_packed_depth_stencil
GL_EXT_texture_lod_bias
GL_EXT_provoking_vertex
GL_EXT_direct_state_access
GL_ARB_color_buffer_float
GL_ARB_shader_image_load_store
GL_ARB_texture_compression_bptc
GL_ARB_texture_compression_rgtc
GL_AMD_interleaved_elements
GL_ARB_draw_instanced
GL_ARB_derivative_control
GL_EXT_draw_buffers2
GL_ARB_viewport_array
GL_ARB_gpu_shader_fp64
GL_ARB_pipeline_statistics_query
GL_EXT_texture_shared_exponent
GL_ARB_draw_elements_base_vertex
GL_ARB_arrays_of_arrays
GL_ARB_vertex_type_10f_11f_11f_rev
GL_AMD_occlusion_query_event
GL_ARB_window_pos
GL_ARB_explicit_attrib_location
GL_EXT_texture_env_dot3
GL_ARB_texture_float
GL_AMD_blend_minmax_factor
GL_ARB_texture_snorm
GL_ARB_point_parameters
GL_ARB_texture_compression
GL_KHR_no_error
GL_AMD_transform_feedback3_lines_triangles
GL_EXT_transform_feedback
GL_ARB_depth_buffer_float
GL_EXT_bindable_uniform
GL_ARB_shader_ballot
GL_EXT_timer_query
GL_ARB_vertex_buffer_object
GL_AMD_pinned_memory
GL_EXT_packed_float
GL_ARB_pixel_buffer_object
GL_EXT_texgen_reflection
WGL_NV_float_buffer
GL_NV_texture_barrier
GL_ATI_envmap_bumpmap
GL_ARB_separate_shader_objects
GL_ARB_buffer_storage
GL_AMD_seamless_cubemap_per_texture
GL_ARB_query_buffer_object
WGL_EXT_framebuffer_sRGB
GL_AMD_draw_buffers_blend
GL_ARB_fragment_program_shadow
GL_EXT_texture_object
WGL_EXT_swap_control
GL_ARB_shading_language_100
GL_ARB_program_interface_query
WGL_ARB_extensions_string
GL_KHR_debug
GL_ARB_compute_shader
GL_KTX_buffer_region
GL_ARB_get_texture_sub_image
GL_ARB_texture_storage
GL_EXT_packed_pixels
GL_EXT_memory_object
GL_EXT_blend_func_separate
GL_ARB_shader_viewport_layer_array
GL_ARB_gpu_shader5
GL_EXT_texture3D
GL_ARB_texture_env_crossbar
GL_AMD_name_gen_delete
GL_EXT_stencil_wrap
GL_ATI_fragment_shader
GL_ARB_ES3_compatibility
GL_EXT_subtexture
GL_ARB_texture_swizzle
GL_ARB_transpose_matrix
GL_EXT_semaphore
GL_EXT_shader_image_load_store
GL_AMD_conservative_depth
GL_EXT_texture_compression_bptc
GL_EXT_texture_compression_latc
GL_EXT_texture_compression_s3tc
GL_EXT_texture_compression_rgtc
GL_ARB_vertex_program
GL_EXT_texture_sRGB
GL_NV_copy_depth_to_color
WGL_EXT_pixel_format_packed_float
WGL_EXT_swap_control_tear
GL_NV_float_buffer
GL_ARB_texture_rectangle
GL_ARB_depth_clamp
GL_ARB_shading_language_packing
GL_ARB_texture_cube_map_array
GL_ARB_framebuffer_sRGB
GL_ARB_sparse_texture
WGL_ARB_render_texture
GL_ARB_texture_env_add
GL_AMD_shader_atomic_counter_ops
GL_ARB_texture_rg
GL_ARB_framebuffer_no_attachments
GL_ARB_blend_func_extended
GL_ATI_texture_float
WGL_EXT_extensions_string
GL_ARB_vertex_attrib_64bit
GL_AMD_shader_trinary_minmax
GL_EXT_vertex_array
GL_EXT_texture_sRGB_decode
GL_ARB_multi_draw_indirect
GL_ARB_texture_gather
GL_ARB_half_float_vertex
GL_SGIS_texture_edge_clamp
GL_ARB_debug_output
GL_EXT_copy_texture
GL_ARB_texture_cube_map
GL_EXT_shadow_funcs
GL_SGIS_texture_lod
GL_AMD_debug_output
GL_EXT_texture_filter_anisotropic
GL_ARB_shader_draw_parameters
GL_ARB_spirv_extensions
GL_EXT_pixel_buffer_object
GL_AMD_framebuffer_sample_positions
GL_EXT_point_parameters
GL_EXT_multi_draw_arrays
WGL_ATI_render_texture_rectangle
GL_ARB_invalidate_subdata
GL_EXT_texture_rectangle
GL_ATI_draw_buffers
GL_ARB_vertex_array_object
GL_ARB_explicit_uniform_location
GL_ARB_shadow
GL_ARB_vertex_array_bgra
GL_NV_copy_image
GL_ARB_shader_group_vote
GL_ARB_compatibility
GL_KHR_parallel_shader_compile
GL_ARB_shader_texture_lod
GL_AMD_gpu_shader_int16
GL_AMD_gpu_shader_int64
GL_SGIS_generate_mipmap
GL_EXT_blend_minmax
WGL_ARB_pixel_format
GL_ARB_shadow_ambient
GL_ARB_texture_query_levels
GL_ARB_point_sprite
GL_ARB_depth_texture
GL_ARB_fragment_layer_viewport
GL_ARB_copy_buffer
WGL_ARB_make_current_read
WGL_ARB_pixel_format_float
GL_KHR_robustness
GL_ARB_indirect_parameters
GL_ARB_bindless_texture
GL_ARB_get_program_binary
GL_ARB_conditional_render_inverted
GL_ARB_transform_feedback2
GL_ARB_transform_feedback3
WGL_ARB_pbuffer
GL_AMD_sparse_texture
GL_EXT_vertex_array_bgra
GL_ARB_vertex_attrib_binding
GL_NV_alpha_to_coverage_dither_control
GL_ARB_map_buffer_range
WGL_AMD_gpu_association
GL_ARB_occlusion_query
WGL_ARB_create_context
GL_ARB_fragment_shader
GL_ARB_draw_buffers
GL_EXT_texture_lod
GL_EXT_vertex_attrib_64bit
GL_ARB_texture_border_clamp
GL_SUN_multi_draw_arrays
GL_AMD_vertex_shader_viewport_index
GL_ARB_copy_image
GL_NV_half_float
GL_ARB_vertex_shader
GL_ARB_texture_mirrored_repeat
GL_EXT_fog_coord
GL_NV_explicit_multisample
GL_AMD_shader_stencil_export
GL_ARB_sparse_buffer
GL_ARB_fragment_program
GL_EXT_framebuffer_blit
GL_EXT_framebuffer_sRGB
GL_ARB_draw_indirect
GL_ARB_imaging
GL_ARB_geometry_shader4
GL_ATI_separate_stencil
GL_ARB_shading_language_420pack
GL_EXT_semaphore_win32
GL_ARB_base_instance
GL_ARB_texture_env_combine
GL_EXT_abgr
GL_EXT_bgra
GL_ARB_texture_non_power_of_two
GL_ARB_texture_buffer_object
GL_EXT_texture_cube_map
GL_AMD_texture_texture4
GL_ARB_texture_barrier
GL_KHR_robust_buffer_access_behavior
GL_EXT_texture_edge_clamp
GL_ARB_seamless_cubemap_per_texture
GL_ARB_sync
GL_ARB_texture_query_lod
GL_ARB_clear_buffer_object
GL_EXT_separate_specular_color
GL_EXT_framebuffer_object
GL_EXT_texture_integer
GL_NV_conditional_render
GL_ARB_draw_buffers_blend
GL_ARB_texture_stencil8
GL_AMD_stencil_operation_extended
GL_EXT_blend_color
GL_ARB_polygon_offset_clamp
GL_ARB_sample_shading
GL_EXT_gpu_program_parameters
GL_ARB_internalformat_query
GL_EXT_texture_mirror_clamp
GL_ARB_timer_query
GL_ARB_cull_distance
GL_NV_texgen_reflection
GL_AMD_query_buffer_object
GL_KHR_context_flush_control
GL_NV_depth_buffer_float
GL_AMD_texture_cube_map_array
GL_ARB_robust_buffer_access_behavior
GL_EXT_blend_subtract
GL_ARB_shader_subroutine
GL_ARB_conservative_depth
GL_EXT_gpu_shader4
WGL_ARB_multisample
GL_ARB_texture_buffer_range
GL_EXT_secondary_color
GL_AMD_gpu_shader_half_float
GL_EXT_texture_storage
GL_ARB_texture_mirror_clamp_to_edge
GL_ARB_transform_feedback_overflow_query
GL_ARB_shader_precision
GL_AMD_gcn_shader
GL_ARB_clip_control
GL_AMD_shader_trace
GL_EXT_texture_swizzle
GL_ATI_texture_mirror_once
WGL_ARB_create_context_no_error
GL_ARB_provoking_vertex
GL_ARB_stencil_texturing
WGL_ARB_context_flush_control
GL_ARB_shader_texture_image_samples
GL_ARB_texture_view
GL_NV_blend_square
GL_ARB_internalformat_query2
GL_OES_EGL_image
GL_EXT_texture_buffer_object
GL_ARB_shader_image_size
WGL_NV_swap_group
GL_EXT_polygon_offset_clamp
GL_EXT_texture_env_combine
GL_EXT_draw_range_elements
GL_EXT_texture_env_add
GL_AMDX_debug_output
GL_AMD_multi_draw_indirect
GL_AMD_vertex_shader_layer
GL_ATI_texture_compression_3dc
GL_ARB_compressed_texture_pixel_storage
GL_ARB_multisample
GL_ARB_shader_objects
GL_ARB_ES3_1_compatibility
WGL_NV_DX_interop
GL_EXT_geometry_shader4
GL_ARB_direct_state_access
GL_AMD_depth_clamp_separate
GL_ARB_texture_env_dot3
GL_NV_primitive_restart
GL_EXT_rescale_normal
GL_EXT_blend_equation_separate
GL_ARB_vertex_type_2_10_10_10_rev
GL_ATI_texture_env_combine3
GL_ARB_shader_storage_buffer_object
GL_EXT_draw_instanced
WGL_ARB_create_context_profile
WGL_I3D_genlock
WGL_ARB_buffer_region
GL_ARB_transform_feedback_instanced
WGL_ATI_pixel_format_float
GL_ARB_texture_buffer_object_rgb32
GL_EXT_histogram
GL_ARB_instanced_arrays
GL_ARB_texture_storage_multisample
GL_IBM_texture_mirrored_repeat
GL_ARB_gl_spirv
GL_ARB_shader_stencil_export
GL_ARB_ES2_compatibility
GL_ARB_uniform_buffer_object
GL_AMD_sparse_texture_pool
GL_EXT_framebuffer_multisample
GL_EXT_copy_buffer
GL_ARB_texture_multisample
GL_ARB_shader_atomic_counters
GL_EXT_shader_integer_mix
GL_EXT_memory_object_win32
GL_ARB_map_buffer_alignment
GL_ARB_sampler_objects
GL_AMD_sample_positions
GL_ARB_occlusion_query2
GL_ARB_framebuffer_object
GL_ARB_enhanced_layouts
GL_ARB_multi_bind
WGL_NV_DX_interop2
GL_AMD_transform_feedback4
GL_EXT_depth_bounds_test
GL_ARB_multitexture
GL_EXT_texture_array
GL_ARB_seamless_cube_map
GL_EXT_texture_snorm
GL_ARB_texture_rgb10_a2ui
GL_ARB_fragment_coord_conventions
GL_ARB_clear_texture
GL_AMD_performance_monitor
GL_ARB_shader_bit_encoding

Core features
v1.1 (100 % - 7/7)
v1.2 (100 % - 8/8)
v1.3 (100 % - 9/9)
v1.4 (100 % - 15/15)
v1.5 (100 % - 3/3)
v2.0 (100 % - 10/10)
v2.1 (100 % - 3/3)
v3.0 (100 % - 23/23)
v3.1 (100 % - 8/8)
v3.2 (100 % - 10/10)
v3.3 (100 % - 10/10)
v4.0 (100 % - 14/14)
v4.1 (100 % - 7/7)
v4.2 (100 % - 13/13)
v4.3 (100 % - 23/23)
v4.4 (100 % - 10/10)
v4.5 (100 % - 11/11)
v4.6 (100 % - 11/11)
vARB 2015 (25 % - 3/12)
OpenGL driver version check (Current: 0, Latest known: ):
 Latest version of display drivers found 
 According the database, you are running the latest display drivers for your video card.
```

**Comment by ZeGentzy** [comment]
Ah thank you, but I was hoping for the info on the "Display mode & pixel" tab.

**Comment by kchibisov** [comment]
![Screenshot (3)](https://user-images.githubusercontent.com/27620401/64478995-59f69d00-d1b9-11e9-847e-cb5cda873fc2.png)

**Comment by ZeGentzy** [comment]
@kchibisov There is more then one pixel format. Can you export a log with all of them (prolly the download button on the top bar does that, but I've never used the app.)

**Comment by kchibisov** [comment]
The app is super confusing btw. I hope this XML will help you.

[report_all.txt](https://github.com/rust-windowing/glutin/files/3586979/report_all.txt)

**Comment by sumit0190** [comment]
@ZeGentzy Aah, it's our good friend `sRGB` at work again!

Since I have access to AMD hardware I was able to reproduce it with latest version of `glutin` and the AMD drivers. Here's what I found.

Let's remove `sRGB` from the picture entirely; i.e., we don't explicitly set `FRAMEBUFFER_SRGB_CAPABLE_EXT` to `0` (since the default pixel format doesn't ask for it). I simply commented out the part where we set `sRGB`, the descriptor then looks like this:

`[8193, 1, 8208, 1, 8211, 8235, 8195, 8231, 8212, 24, 8219, 8, 8226, 24, 8227, 8, 8209, 1, 8210, 0, 0]`

**This works correctly**.

Now I set `sRGB` to `1`, even though `pf_reqs` doesn't ask for it. The descriptor then looks like this:

`[8193, 1, 8208, 1, 8211, 8235, 8195, 8231, 8212, 24, 8219, 8, 8226, 24, 8227, 8, 8209, 1, 8210, 0, 8361, 1, 0]`

**This also works correctly**. Notice the additional field `8361`, which corresponds to my `sRGB` support.

Now we come to the crux; I try to set `sRGB` to `0`, which is what `pf_reqs` has. The descriptor looks like this:

`[8193, 1, 8208, 1, 8211, 8235, 8195, 8231, 8212, 24, 8219, 8, 8226, 24, 8227, 8, 8209, 1, 8210, 0, 8361, 0, 0]`

**THIS DOESN'T WORK - the pixel format isn't found**. Again, just to verify that `sRGB` really is the culprit, I set the descriptor to

`[8361, 0, 0]`

and again, **it doesn't work**. Now I am a little confused, as it seems that the `wglChoosePixelFormatARB` is simply unable to find a format that does not have `sRGB` . So I fire up my GLView and guess what I find? 

**There isn't a single integer format that doesn't have `sRGB`**. I literally tried all of them to confirm this. Only floating-point formats seem to have a few where `sRGB` is absent.
